servicetalk-opentracing-zipkin-publisher/src/main/java/io/servicetalk/opentracing/zipkin/publisher/reporter/HttpReporter.java

```diff
@@ -0,0 +1,287 @@
+        } else {
+            // As we send maxConcurrentReports number of parallel requests, each with roughly batchSizeHint number of
+            // spans, we hold a maximum of that many Spans in-memory that we can send in parallel to the collector.
+            buffer = newPublisherProcessorDropHeadOnOverflow(builder.batchSizeHint * builder.maxConcurrentReports);
+            spans = fromSource(buffer)
+                    .buffer(forCountOrTime(builder.batchSizeHint, builder.maxBatchDuration,
+                            () -> new ListAccumulator(builder.batchSizeHint), client.executionContext().executor()))
+                    .filter(accumulate -> !accumulate.isEmpty())
+                    .map(bufferedSpans -> allocator.wrap(spanEncoder.encodeList(bufferedSpans)));
+        }
+
+        final CompletableSource.Processor spansTerminated = newCompletableProcessor();
+        toSource(spans.flatMapCompletable(encodedSpansReporter(client, builder.codec), builder.maxConcurrentReports))
+                .subscribe(spansTerminated);
+
+        closeable.prepend(toAsyncCloseable(graceful -> {
+            closeInitiated = true;
+            try {
+                buffer.onComplete();
+            } catch (Throwable t) {
+                LOGGER.error("Failed to dispose request buffer. Ignoring.", t);
+            }
+            return graceful ? fromSource(spansTerminated) : completed();
+        }));
+
+        return buffer;
+    }
+
+    private static Function<Buffer, Completable> encodedSpansReporter(final HttpClient client, final Codec codec) {
+        final String path;
+        final CharSequence contentType;
+        switch (codec) {
+            case JSON_V1:
+                path = V1_PATH;
+                contentType = APPLICATION_JSON;
+                break;
+            case JSON_V2:
+                path = V2_PATH;
+                contentType = APPLICATION_JSON;
+                break;
+            case THRIFT:
+                path = V2_PATH;
+                contentType = THRIFT_CONTENT_TYPE;
+                break;
+            case PROTO3:
+                path = V2_PATH;
+                contentType = PROTO_CONTENT_TYPE;
+                break;
+            default:
+                throw new IllegalArgumentException("Unknown codec: " + codec);
+        }
+        return encodedSpans -> client.request(
+                client.post(path).addHeader(CONTENT_TYPE, contentType).payloadBody(encodedSpans))
+                .ignoreElement()
+                .onErrorResume(cause -> {
+                    LOGGER.error("Failed to send a span, ignoring.", cause);
+                    return completed();
+                });
+    }
+
+    /**
+     * A builder to create a new {@link HttpReporter}.
+     */
+    public static final class Builder {
+        private Codec codec = Codec.JSON_V2;
+        private final SingleAddressHttpClientBuilder<?, ?> clientBuilder;
+        private boolean disableBatching;
```

Comment:
optional: consider removing this additional state and set `batchSizeHint < 0` to indicate disabled

Reply:
Since this is a builder I would prefer being explicit as opposed to overloading a field.